tools/ARIAtools/product.py

```diff
@@ -74,6 +96,126 @@ def package_dict(scene, new_scene, scene_ind,
     return new_dict
 
 
+def remove_scenes(products):
+    """
+    For ARIA-S1 GUNW products, identify and reject legacy v2 products if
+    in the presence of v3 products for a given IFG.
+
+    'products' = list of input products
+    """
+    track_legacy_products = []
+    sorted_products = []
+
+    # only check ARIA-S1 GUNW products
+    if os.path.basename(products[0][1]['productBoundingBox'].split( \
+        '"')[1])[:2] == 'S1':
+        for i in enumerate(products[:-1]):
+            scene = i[1]
+            new_scene = products[i[0] + 1]
+            scene_t_ref = datetime.datetime.strptime(
+                scene[0]['pair_name'][:8], "%Y%m%d")
+            new_scene_t_ref = datetime.datetime.strptime(
+                new_scene[0]['pair_name'][:8], "%Y%m%d")
+            scene_t = datetime.datetime.strptime(
+                scene[0]['pair_name'][9:], "%Y%m%d")
+            new_scene_t = datetime.datetime.strptime(
+                new_scene[0]['pair_name'][9:], "%Y%m%d")
+
+            # Only pass scene if it temporally (i.e. in same orbit)
+            # overlaps with reference scene
+            if abs(new_scene_t - scene_t) <= ONE_DAY and \
+                abs(new_scene_t_ref - scene_t_ref) <= ONE_DAY:
+
+                # Check if IFG dict corresponding to ref prod already exists
+                # and if it does then append values
+                try:
+                    dict_ind = sorted_products.index(next(
```

Comment:
IMO this list comp is very confusing I can't tell what is being done here without significant mental effort. same comment on other list comps below

Reply:
![john-mclane](https://github.com/aria-tools/ARIA-tools/assets/15522503/35ad0ce6-4cf2-4378-acf3-98d38dc805c7)